src/main/java/com/bunq/sdk/http/BunqBasicHeader.java

```diff
@@ -1,26 +1,47 @@
 package com.bunq.sdk.http;
 
+import okhttp3.Response;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
 public class BunqBasicHeader {
-  private String name;
-  private String value;
+  private static final String DELIMITER_HEADER_NAME_AND_VALUE = ": ";
+  private static final String NEWLINE = "\n";
+
+  private final BunqHeader name;
+  private final String value;
+
+  public static BunqBasicHeader get(BunqHeader header,Response response) {
+    return new BunqBasicHeader(header, response.header(header.getHeader()));
+  }
 
-  /**
-   */
-  public BunqBasicHeader(String name, String value) {
+  public static Optional<BunqBasicHeader> get(String header, String value) {
+    return BunqHeader.parse(header).map(h -> new BunqBasicHeader(h,value));
```

Comment:
This is not java 7 compatible. Please refactor this so that it is. 

And please add missing space after `,`